src/rust/plugin-work-queue/migrations/1640037910_plugin_registry_migration.sql

```diff
@@ -0,0 +1,86 @@
+    tenant_id            uuid        NOT NULL,
+    plugin_id            uuid        NOT NULL,
+    pipeline_message     bytea       NOT NULL,
+    execution_result     bytea,
+    status               status_t    NOT NULL,
+    creation_time        timestamptz NOT NULL,
+    last_updated         timestamptz NOT NULL,
+    visible_after        timestamptz,
+    try_count            integer     NOT NULL,
+    trace_id             uuid        NOT NULL,
+    CHECK (length(pipeline_message) < megabytes(256)),
+    CHECK (length(execution_result) < megabytes(256)),
+    CHECK (last_updated >= creation_time)
+)
+    PARTITION BY RANGE (creation_time);
+
+CREATE TABLE IF NOT EXISTS plugin_work_queue.analyzer_plugin_executions
+(
+    execution_key        bigserial   NOT NULL,
+    tenant_id            uuid        NOT NULL,
+    plugin_id            uuid        NOT NULL,
+    pipeline_message     bytea       NOT NULL,
+    execution_result     bytea,
+    status               status_t    NOT NULL,
+    creation_time        timestamptz NOT NULL,
+    last_updated         timestamptz NOT NULL,
+    visible_after        timestamptz,
+    try_count            integer     NOT NULL,
+    trace_id             uuid        NOT NULL,
+    CHECK (length(pipeline_message) < megabytes(256)),
+    CHECK (length(execution_result) < megabytes(256)),
+    CHECK (last_updated >= creation_time)
+)
+    PARTITION BY RANGE (creation_time);
+
+CREATE INDEX IF NOT EXISTS execution_key_ix ON plugin_work_queue.generator_plugin_executions (execution_key);
+CREATE INDEX IF NOT EXISTS creation_time_ix ON plugin_work_queue.generator_plugin_executions (creation_time);
+
+CREATE INDEX IF NOT EXISTS execution_key_ix ON plugin_work_queue.analyzer_plugin_executions (execution_key);
+CREATE INDEX IF NOT EXISTS creation_time_ix ON plugin_work_queue.analyzer_plugin_executions (creation_time);
+
+CREATE SCHEMA partman;
+CREATE EXTENSION pg_partman WITH SCHEMA partman;
+
+
+-- TODO: Check optimize_contraint
+SELECT partman.create_parent(p_parent_table => 'plugin_work_queue.generator_plugin_executions',
+                             p_control => 'creation_time',
+                             p_type => 'native',
+                             p_interval=> 'daily',
+                             p_premake => 30
+           );
+
+SELECT partman.create_parent(p_parent_table => 'plugin_work_queue.analyzer_plugin_executions',
+                             p_control => 'creation_time',
+                             p_type => 'native',
+                             p_interval=> 'daily',
+                             p_premake => 30
+           );
+
+CREATE EXTENSION pg_cron;
+
+UPDATE partman.part_config
+SET infinite_time_partitions = true,
+    retention                = '1 month',
+    retention_keep_table= true
+WHERE (
+    parent_table = 'plugin_work_queue.generator_plugin_executions' OR
+    parent_table = 'plugin_work_queue.analyzer_plugin_executions'
+);
```

Comment:
been a hot second since i sqld but
where parent_table in ('first', 'second') ?

Reply:
Is that preferable? I think that would work fwiw but I have no sense of whether one is better or worse.

Reply:
i'd imagine perf wise they're the same